the-collab-lab/tcl-47-smart-shopping-list
# Implement MUI UI Library, adjust theme, and add checkbox

## Description

This issue introduces the MUI UI framework to the project. It also establishes a theme and implements a few MUI components including the Checkbox and a search bar that utilizes icon adornment. 

## Related Issue

closes #31 

## Acceptance Criteria

- [x] [Install MUI and related dependencies](https://mui.com/material-ui/getting-started/installation/#npm)
`npm install @mui/material @emotion/react @emotion/styled
`
- [x] Install Roboto (or font of your choosing) using Google Fonts CDN, by adding the following to the project `<Head>` tag: 
`<link
  rel="stylesheet"
  href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700&display=swap"
/>` and ensure that it is working properly.

- [x] Install Material Icon font
`npm install @mui/icons-material` and ensure that you can successfully add an icon. Note: we will eventually want the search icon and a delete icon.
- [x] Replace checkmark in ListItem with MUI checkmark component

### A Stretch Goal
Implement a basic theme. The checkmark is only used in this file, however, buttons and other UI elements that are used in multiple places, ideally have style overrides set in one place rather than in each instance.
- [x] Make `theme.js` file in `src/themes`
- [x] Export basic theme changing primary color to the primary green used throughout the project. 

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
| ✓    | :link: Update dependencies |
|     | :scroll: Docs              |

## Updates

### Before

<img width="693" alt="Before" src="https://user-images.githubusercontent.com/49693755/202594678-e4f793f0-2cdc-4cdb-97e8-c7024dae9398.png">

### After

<img width="686" alt="After" src="https://user-images.githubusercontent.com/49693755/202594751-a6b38c51-6bc3-4821-b193-fc6a0d55d405.png">

## Testing Steps / QA Criteria

1. Git Pull WN-MUI-Integration
2. run `npm i` in terminal to install new dependencies 
3. run `npm start` to ensure that the UI changes reflect the 'After' image shown above
4. Start having fun with MUI!
